tree/dataframe/inc/ROOT/RCsvDS.hxx

```diff
@@ -95,8 +98,11 @@ public:
 ///                        (default `true`).
 /// \param[in] delimiter Delimiter character (default ',').
 /// \param[in] linesChunkSize bunch of lines to read, use -1 to read all
+/// \param[in] colTypes Allow user to specify custom column types, accepts an unordered map with keys being
+///                      column type, values being type alias ('b' for boolean, 'd' for double, 'l' for
+///                      Long64_t, 's' for std::string)
```

Comment:
Well, that is because there is already a map of the possible types in `RCsvDS.cxx:107`. As this map defined in  `RCsvDS.cxx:107` was always internal and never exposed to users, we can simply change the keys there to make it consistent with TBranch.cxx. How does this sound?

Reply:
sounds good to me (@etejedor ?). For strings we should check what notation is used in CSV headers that are parsed by `TTree::ReadFile` (@pcanal ?)

Reply:
A note here: in `TBranch.cxx:148` we have `s` : a 16 bit unsigned integer (`UShort_t`) and `S` : a 16 bit signed integer (`Short_t`). We currently do not support any `short` type, but we should be carefult not to create ambiguity with strings and shorts.

Reply:
that's more of the same problem, `l` for `Long64_t` is also confusing because in TTree `l` is `ULong64_t`

Reply:
Well, that would not change the behaviour of the csv dataframe, as said - this is not exposed. It easy to change what is there already - `Long64_t, double, bool`. The only more interesting question is what character should map to `std::string`? Since `O` is used for bool (as being the second letter of `bool`) - my proposal would be to pick `T` for strings.

Reply:
@pcanal as far as we can tell TTree::ReadFile does not support strings in the CSV data, can we just pick `T` as the branch type descriptor then?

Reply:
I think it supports then as `const char*` via `C`.  `T` is indeed 'available'.

I also agree that we should **not** deviate from the convention used by TTree's leaflist.

Reply:
However this puts us in a bit of a pickle because RCsvDS does not support sub-collections, right? so we can't say a string is a column that contains sub-collections of chars

Reply:
First question is right - RCsvDS supports now only longs, doubles, bools, strings.
We might keep the `C` available for chars collections, if/when we start supporting that as well.

By similar token, if we now use `C` for strings inside RCsvDS --> If for some reason TTree starts supporting strings with some letter (which will certainly be not `C`), there is going to be mismatch between TTree and RCsvDS which we don't want.